### ISSUE TITLE
Extension is crashing without user interaction

### ISSUE THREAD

**Issue by chevdor** [polkadot-js/extension#468]
Unfortunately, I don't have much of an error as the console of the background page closes as soon as the crash occurs.
I will try providing info, but I don't know what is relevant or not.

Here is what I did:
- I am mostly playing on https://app.subsocial.network/spaces/all
- Loading the site, I need to accept the usage of the ext. OK
- I did use the "remember me 15min" checkbox and used subsocial for about 15min
- I am now at the point where, when trying to edit the description of a space in subsocial, I need to sign a tx, the extension opens and crashes even without me doing anything

I need to test further and the issue may be a subsocial issue when the description contains a specific content (I am suspecting length or content such as http:// links may cause troubles).

**Comment by chevdor** [comment]
![image](https://user-images.githubusercontent.com/738724/94568242-8a015a00-026c-11eb-80c6-f529583669dc.png)

while editing the `srtool` space at https://app.subsocial.network/spaces/@srtool/edit
as:
![image](https://user-images.githubusercontent.com/738724/94568368-af8e6380-026c-11eb-8119-88df5bb73d88.png)

I had changed the 3 fields showing a green check mark.

I do suspect that the issue occurs only after the 15min.

**Comment by chevdor** [comment]
Clicking reload fails as well, this time with a console error:
```
Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.
```

**Comment by chevdor** [comment]
I managed to get some logs which are also not very helpful:
```
[19107:20227:0929/193924.452007:VERBOSE1:network_delegate.cc(32)] NetworkDelegate::NotifyBeforeURLRequest: http://127.0.0.1:5001/api/v0/swarm/peers
[19089:775:0929/193924.471731:INFO:CONSOLE(2)] "Error while ipfs.swarm.peers: TypeError: Failed to fetch", source: chrome-extension://nibjojkomfdiaoajekhjakgkdhaomnch/dist/bundles/backgroundPage.bundle.js (2)
[19921:775:0929/193925.677775:FATAL:memory.cc(40)] Out of memory. size=0
[19123:775:0929/193925.949478:VERBOSE1:script_context.cc(126)] Destroyed context for extension
  extension id: mopnmbcafieddcagagdcbnhejhlodfdd
  effective extension id: mopnmbcafieddcagagdcbnhejhlodfdd
[19123:775:0929/193925.949557:VERBOSE1:script_context.cc(126)] Destroyed context for extension
  extension id: 
  effective extension id: 
```

At least I now know how to reproduce the issue 100%. Unfortunately, it will likely be visible by me as this example fails in a single subsocial space: https://app.subsocial.network/spaces/1214/edit setting the description to:

```
FAIL
Substrate Runtime Toolbox (srtool) verify substrate runtimes.

OK
Substrate Runtime Toolbox (srtool) verify substrate runtimes

OK
Substrate Runtime Toolbox (srtool) verify substrate.
```

**Comment by jacogr** [comment]
So I think we need to add the ErrorBoundary (linked issue), so it (hopefully) _doesn't_ crash. Then run the same scenario through. However, the OOM, well, all bets may be off...

**Comment by Tbaut** [comment]
I'll work on #469  this tomorrow :crossed_fingers:

**Comment by Tbaut** [comment]
Forgot to log here the outcome of yesterday.

Some more info:
- Happening on the latest Brave.
- The crash happens before clicking anything on the Popup
- I'm still not able to reproduce.
- problem still happening with the build with ErrorBoundaries from #469 which shows that the problem is not in React. The screenshots from @chevdor right before it crashes:

![image](https://user-images.githubusercontent.com/33178835/94817413-4427cb80-03fd-11eb-9376-4ab0e6dd40e9.png)

`pri(metadata.get)` is the last call from the Signer.
This call happens in the Extrinsic component from the Popup here: https://github.com/polkadot-js/extension/blob/master/packages/extension-ui/src/Popup/Signing/Extrinsic.tsx#L104

It seems to return correctly since there's the [in], the [out], and the info on the extrinsic like `method.data` are also displayed.

Tbh I'm not sure which direction we should check.

**Comment by jacogr** [comment]
Ok, the ErrorBoundary is a good-thing-to-have(tm) anyway, we have had white pages before... So it does OOM, the last call is the metadata, most probably there is something wrong with it. Would suggest the following -

- manually remove the metadata storage entry (it is in "normal localStorage" for the extension)
- see if it persists

Then -

- update the metadata from the apps UI
- see if it happens or is ok

My guess would be to look where the parsing happens, but that goes down the tree. Have seen "stack exhausted" errors on these before, but never metadata.

**Comment by Tbaut** [comment]
Thanks for the pointers. @chevdor did mention that he had update the metadata in apps, I hadn't.
I can still not reproduce when I do update them as well.

The metadata seem to be in the background local storage though, so we can't manually delete them. I'll make a custom build for @chevdor to be able to delete them and test.

**Comment by jacogr** [comment]
Ahh, I have the Storage Explorer extension on chrome, gives insight into these. My guess atm would be that the strings are being wrongly decoded (some type is wrong before that). Unlike the Vec types, not sure if strings complain when it gets a too-large (read: bullshit) value.

**Comment by Tbaut** [comment]
Sending in DM a build from this branch, with a button to delete the subsocial metadata : https://github.com/polkadot-js/extension/compare/tbaut-delete-subsocial-metadata?expand=1

**Comment by chevdor** [comment]
Here are some updates:
- I tried to reproduce the issue using another machine => it all worked fine (no crash)
- I went back to the 'faulty' machine and tested again using the 0.35.0 "packed" version from @Tbaut => it worked ( no crash)
- I switched to the "stock" version (0.34.1) => it worked this time ! 

So in short, I can no longer reproduce the issue.
I would propose to close and reopen if I run into the issue again.
I leave open for now in case you still want to had some checks but feel free to close if you agree.
